provenance/provenance.sol

```diff
@@ -0,0 +1,55 @@
+// provenance concept contract created by M.ElBoudi, A.Blackwell, M.Terzi et al during Ethereum development workshop in London
+
+contract Certifiers {
+        mapping (address => bool) certifier;
+        address provenance;
+        
+        //initialiser is run the first time the contract is uploaded to the network. Launched by Provenance and reads provenance "public key"
+        function Certifiers () {
+            provenance = msg.sender;    
+        }
+        
+        function createNewCertifier() returns (bool completed) {
+                certifier[msg.sender] = false;
+                return true;
+        }
+        
+        //if Provenance is launching the contract, then the entity becomes certified
+        function certify(address targetCertifier) returns (bool completed) {
+            if (provenance == msg.sender) {
+                certifier[targetCertifier] = true;
+                return true;
+            } else {
+                return false;    
+            }
+        }
+}
+
+contract Producers{
+    struct meta{
+        bytes32 description;
```

Comment:
Is it important to have this structure (description, phoneNumber, name) inside the contract? You could just use a `string data`, which will provide storage for a dynamically-sized string which can contain e.g. json-formatted data whose structure can be extended without changing the contract.